test/functional/api/cas/casadm.py

```diff
@@ -54,7 +52,9 @@ def add_core(cache: Cache, core_dev: Device, core_id: int = None, shortcut: bool
                      core_id=_core_id, shortcut=shortcut))
     if output.exit_code != 0:
         raise CmdException("Failed to add core.", output)
-    return Core(core_dev.system_path, cache.cache_id)
+    core = Core(core_dev.system_path, cache.cache_id)
+    core.filesystem = core_dev.filesystem
```

Comment:
You can add a method in the Device class which will discover a filesystem on a device and using this method you can fill the filesystem field in both Device and Core classes.